sys/riscv/riscv/pmap.c

```diff
@@ -2009,6 +2012,14 @@ pmap_remove_kernel_l2(pmap_t pmap, pt_entry_t *l2, vm_offset_t va)
 	    __func__, l2, oldl2));
 }
 
+static void
+pmap_page_dirty(pt_entry_t tpte, vm_page_t m)
+{
+	vm_page_dirty(m);
+	if ((tpte & (PTE_SC | PTE_CD)) == (PTE_SC | PTE_CD))
+		vm_page_capdirty(m);
```

Comment:
Why does dirty tracking not care about W but capdirty cares about SC (CW)? I'd expect this to just be a `(tpte & PTE_CD) != 0` check as the capability equivalent.

Reply:
I've added a comment that this is, as with almost all uses of `PTE_CD` right now, a hack to work with hardware that doesn't understand that `PTE_SC | PTE_CD` is the gating condition for stores and instead permit `PTE_SC` without `PTE_CD` to store.